packages/next/src/build/index.ts

```diff
@@ -1276,6 +1301,94 @@ export default async function build(
         app: appPaths.length > 0 ? appPaths : undefined,
       }
 
+      await nextBuildSpan
+        .traceChild('generate-route-types')
+        .traceAsyncFn(async () => {
+          const routeTypesFilePath = path.join(distDir, 'types', 'routes.d.ts')
+          await mkdir(path.dirname(routeTypesFilePath), { recursive: true })
```

Comment:
```suggestion
          await fs.mkdir(path.dirname(routeTypesFilePath), { recursive: true })
```

We use `fs.mkdir` elsewhere here so ideally we match that.